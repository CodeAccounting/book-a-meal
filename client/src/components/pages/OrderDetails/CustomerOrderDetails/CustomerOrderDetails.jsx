import React, { Component, Fragment } from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { userPropTypes, catererOrderObjPropTypes } from '../../../../helpers/proptypes';
import View from '../../../shared/View';
import OrderSummary from '../../../shared/OrderSummary';
import OrderAmount from '../../../shared/OrderAmount';

/**
 * @exports
 * @class CustomerOrderDetails
 * @extends Component
 * @classdesc Creates CustomerOrderDetails Component
 * @returns {JSX} CustomerOrderDetails Component
 */
class CustomerOrderDetails extends Component {
  static propTypes = {
    ...userPropTypes,
    order: catererOrderObjPropTypes,
    isFetching: PropTypes.bool.isRequired,
    logout: PropTypes.func.isRequired,
    fetchOrders: PropTypes.func.isRequired,
  }

  static defaultProps = {
    order: null
  }

  /**
   * @memberof CustomerOrderDetails
   * @returns {JSX} CustomerOrderDetails Component
  */
  componentDidMount() {
    this.fetchOrders();
  }

  /**
   * @memberof CustomerOrderDetails
   * @returns {JSX} CustomerOrderDetails Component
  */
  fetchOrders = () => {
    this.props.fetchOrders();
  }

  /**
   * @memberof CustomerOrderDetails
   * @param {object} order
   * @param {func} deliverOrder
   * @returns {JSX} CustomerOrderDetails Component
   */
  renderCustomerDetails = (order) => {
    const status = order.status === 'started' ? 'pending' : order.status;

    const statusClass = classNames({
      warning: status === 'pending',
      success: status === 'delivered',
      danger: status === 'canceled'
    });

    return (
      <div className="customer-details">
        <p>Status:
          <span className={statusClass} style={{ textTransform: 'capitalize' }}> {status}</span>
        </p>
        <p>Address Provided: {order.deliveryAddress}</p>
        <p>Number Provided: {order.deliveryPhoneNo}</p>
      </div>
    );
  }

  /**
   * @memberof CustomerOrderDetails
   * @returns {JSX} CustomerOrderDetails Component
  */
  renderControlBtns = () => (
    <Fragment>
      <br />
      <div className="d-flex-row control-btns">
        <a href="./order-confirmation.html">
          <button className="btn btn-sec order-details-btn">Edit</button>
        </a>
        <a href="./user-menu.html">
          <button className="btn btn-sec-danger order-details-btn">Cancel</button>
        </a>
      </div>
    </Fragment>
  )

  /**
   * @memberof CustomerOrderDetails
   * @returns {JSX} CustomerOrderDetails Component
  */
  renderDetails = () => {
    const { order } = this.props;

    return (
      <Fragment>
        <OrderSummary meals={order.meals} />
        <OrderAmount meals={order.meals} />
        {this.renderCustomerDetails(order)}
        {order.status === 'started' && this.renderControlBtns()}
      </Fragment>
    );
  }

  /**
   * @memberof CustomerOrderDetails
* @returns {JSX} CustomerOrderDetails Component
    */
  renderCustomerOrderDetails = () => {
    const { order } = this.props;

    return (
      <Fragment>
        <div className="order-confirmation order-details">
          <h2 className="text-center" style={{ paddingTop: '1em' }}>Order #{order.id}</h2>
          <p className="order-date text-center">{moment().format('YY/MM/DD h:mm a')}</p>
          <hr />
          {this.renderDetails()}
        </div>
      </Fragment>
    );
  }

  /**
   * @memberof CustomerOrderDetails
   * @returns {JSX} CustomerOrderDetails Component
  */
  render() {
    const { user, logout, isFetching } = this.props;

    return (
      <View user={user} logout={logout} type="customerOrderDetails" isFetching={isFetching}>
        <Fragment>
          {!this.props.order && <p className="text-center">This Order Does Not Exist</p>}
          {this.props.order && this.renderCustomerOrderDetails()}
        </Fragment>
      </View>
    );
  }
}

export default CustomerOrderDetails;
