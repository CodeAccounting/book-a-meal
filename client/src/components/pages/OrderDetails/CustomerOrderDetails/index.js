import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import { bindActionCreators } from 'redux';
import CustomerOrderDetails from './CustomerOrderDetails';
import { editOrder, cancelOrder } from '../../../../store/operations/orders';
import { fetchOrder } from '../../../../store/operations/singleOrder';
import { logout } from '../../../../store/operations/auth';

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  order: state.singleOrder.item,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchOrder, logout, push, editOrder, cancelOrder
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(CustomerOrderDetails);
