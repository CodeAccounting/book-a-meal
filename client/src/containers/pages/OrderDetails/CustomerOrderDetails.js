import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import CustomerOrderDetails from '../../../components/pages/OrderDetails/CustomerOrderDetails';
import { editOrder, cancelOrder } from '../../../actions/orders';
import { fetchOrder } from '../../../actions/singleOrder';
import { logout } from '../../../actions/auth';

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  order: state.singleOrder.item,
});

export default connect(mapStateToProps, {
  fetchOrder, logout, push, editOrder, cancelOrder
})(CustomerOrderDetails);
