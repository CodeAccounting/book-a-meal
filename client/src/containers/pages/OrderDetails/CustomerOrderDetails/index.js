import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import { bindActionCreators } from 'redux';
import CustomerOrderDetails from '../../../../components/pages/OrderDetails/CustomerOrderDetails';
import { editOrder, cancelOrder } from '../../../../actions/orders';
import { fetchOrder } from '../../../../actions/singleOrder';
import { logout } from '../../../../actions/auth';

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  order: state.singleOrder.item,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchOrder, logout, push, editOrder, cancelOrder
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(CustomerOrderDetails);
