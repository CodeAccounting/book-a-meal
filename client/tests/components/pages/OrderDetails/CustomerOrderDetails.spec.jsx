import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import CustomerOrderDetails from '../../../../src/components/pages/OrderDetails/CustomerOrderDetails/CustomerOrderDetails';
import ConnectedCustomerOrderDetails from '../../../../src/components/pages/OrderDetails/CustomerOrderDetails';
import { customer, customersOrdersObj, initialValues } from '../../../setup/data';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  ...initialValues
});

const { now } = Date;

describe('OrderDetails', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 0);
  });

  afterAll(() => {
    Date.now = now;
  });

  it('renders correctly when not fetching', () => {
    const wrapper = shallow(<CustomerOrderDetails
      user={customer}
      logout={jest.fn()}
      fetchOrder={jest.fn()}
      editOrder={jest.fn()}
      cancelOrder={jest.fn()}
      push={jest.fn()}
      order={customersOrdersObj.orders[0]}
      isFetching={false}
      match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa70c25' } }}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Preloader when fetching', () => {
    const wrapper = shallow(<CustomerOrderDetails
      user={customer}
      logout={jest.fn()}
      fetchOrder={jest.fn()}
      editOrder={jest.fn()}
      cancelOrder={jest.fn()}
      push={jest.fn()}
      order={customersOrdersObj.orders[0]}
      isFetching
      match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa70c25' } }}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('Preloader')).toBeTruthy();
  });

  it('renders message when not fetching and that order doesn\'t exist', () => {
    const comp = (
      <Provider store={store}>
        <ConnectedCustomerOrderDetails
          user={customer}
          isFetching={false}
          match={{ params: { id: 'fb097b-5959-45ff-8e21-51184fa61c25' } }}
        />
      </Provider>);

    const wrapper = mount(comp, rrcMock.get());

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('p.text-center').text()).toEqual('This Order Does Not Exist');
  });

  it('renders delivered order', () => {
    const newStore = mockStore({
      ...initialValues,
      singleOrder: { ...initialValues.singleOrder, item: customersOrdersObj.orders[1] }
    });

    const comp = (
      <Provider store={newStore}>
        <ConnectedCustomerOrderDetails
          user={customer}
          match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa60c25' } }}
        />
      </Provider>);

    const wrapper = mount(comp, rrcMock.get()).find(CustomerOrderDetails);

    expect(wrapper.find('.order-status').text()).toEqual(' delivered');
  });

  it('renders canceled order', () => {
    const newStore = mockStore({
      ...initialValues,
      singleOrder: { ...initialValues.singleOrder, item: customersOrdersObj.orders[0] }
    });

    const comp = (
      <Provider store={newStore}>
        <ConnectedCustomerOrderDetails
          user={customer}
          match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa70c25' } }}
        />
      </Provider>);

    const wrapper = mount(comp, rrcMock.get()).find(CustomerOrderDetails);

    expect(wrapper.find('.danger').text()).toEqual(' canceled');
  });

  it('renders pending order', () => {
    const newStore = mockStore({
      ...initialValues,
      singleOrder: { ...initialValues.singleOrder, item: customersOrdersObj.orders[2] }
    });

    const comp = (
      <Provider store={newStore}>
        <ConnectedCustomerOrderDetails
          user={customer}
          order={customersOrdersObj.orders[2]}
          match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa80c25' } }}
        />
      </Provider>);

    const wrapper = mount(comp, rrcMock.get()).find(CustomerOrderDetails);
    expect(wrapper.find('.warning').text()).toEqual(' pending');
  });

  it('renders started order and shows control buttons', () => {
    const newStore = mockStore({
      ...initialValues,
      singleOrder: { ...initialValues.singleOrder, item: customersOrdersObj.orders[3] }
    });

    const comp = (
      <Provider store={newStore}>
        <ConnectedCustomerOrderDetails
          user={customer}
          match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa90c25' } }}
        />
      </Provider>);

    const wrapper = mount(comp, rrcMock.get()).find(CustomerOrderDetails);
    expect(wrapper.find('.order-status').text()).toEqual(' pending');
    expect(wrapper.find('.d-flex-row.control-btns').length).toBeTruthy();
  });

  it('renders connected component', () => {
    const newStore = mockStore({
      ...initialValues,
      singleOrder: { ...initialValues.singleOrder, item: customersOrdersObj.orders[1] }
    });

    const comp = (
      <Provider store={newStore}>
        <ConnectedCustomerOrderDetails
          user={customer}
          isFetching={false}
          match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa60c25' } }}
        />
      </Provider>);

    const wrapper = mount(comp, rrcMock.get());

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('calls edit order method on edit button click', () => {
    const newStore = mockStore({
      ...initialValues,
      singleOrder: { ...initialValues.singleOrder, item: customersOrdersObj.orders[3] }
    });

    const comp = (
      <Provider store={newStore}>
        <ConnectedCustomerOrderDetails
          user={customer}
          isFetching={false}
          match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa90c25' } }}
        />
      </Provider>);

    const wrapper = mount(comp, rrcMock.get());

    wrapper.find(CustomerOrderDetails).find('.control-btns>a').at(0).simulate('click');

    const storedOrder = JSON.parse(localStorage.getItem('bookamealorder'));

    expect(storedOrder.order.id).toEqual('fb097bde-5959-45ff-8e21-51184fa90c25');

    wrapper.unmount();
  });

  it('calls cancel order method on cancel button click', () => {
    const cancelOrderMock = jest.fn();
    const newStore = mockStore({
      ...initialValues,
      singleOrder: { ...initialValues.singleOrder, item: customersOrdersObj.orders[0] }
    });

    const comp = (
      <Provider store={newStore}>
        <CustomerOrderDetails
          user={customer}
          logout={jest.fn()}
          fetchOrder={jest.fn()}
          editOrder={jest.fn()}
          cancelOrder={cancelOrderMock}
          push={jest.fn()}
          isFetching={false}
          order={customersOrdersObj.orders[3]}
          match={{ params: { id: 'fb097bde-5959-45ff-8e21-51184fa90c25' } }}
        />
      </Provider>);

    const wrapper = mount(comp, rrcMock.get());

    wrapper.find(CustomerOrderDetails).find('.control-btns>a').at(1).simulate('click');

    expect(cancelOrderMock).toHaveBeenCalled();

    wrapper.unmount();
  });
});
