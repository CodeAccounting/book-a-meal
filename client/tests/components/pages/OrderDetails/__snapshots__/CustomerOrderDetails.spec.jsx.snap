// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetails renders Preloader when fetching 1`] = `
<View
  isFetching={true}
  logout={[MockFunction]}
  showTime={true}
  type="customerOrderDetails"
  user={
    Object {
      "email": "emiola@olasanmi.com",
      "firstname": "Emiola",
      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
      "lastname": "Olasanmi",
      "role": "customer",
    }
  }
>
  <React.Fragment>
    <React.Fragment>
      <div
        className="order-confirmation order-details"
      >
        <h2
          className="text-center"
          style={
            Object {
              "paddingTop": "1em",
            }
          }
        >
          Order #
          fb097bde-5959-45ff-8e21-51184fa70c25
        </h2>
        <p
          className="order-date text-center"
        >
          70/01/01 12:00 am
        </p>
        <hr />
        <React.Fragment>
          <OrderSummary
            meals={
              Array [
                Object {
                  "caterer": Object {
                    "businessAddress": "4, Church Street, Yaba",
                    "businessName": "BellyFill",
                    "businessPhoneNo": "+2348134567890",
                    "email": "belly@fill.com",
                  },
                  "delivered": false,
                  "description": "",
                  "id": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
                  "imageURL": "http://www.gratednutmeg.com/wp-content/uploads/2015/03/DSC_07722.jpg",
                  "price": "2000.00",
                  "quantity": 2,
                  "title": "Adalu (Beans and Corn) and Local Fish ",
                  "vegetarian": false,
                },
              ]
            }
          />
          <OrderAmount
            meals={
              Array [
                Object {
                  "caterer": Object {
                    "businessAddress": "4, Church Street, Yaba",
                    "businessName": "BellyFill",
                    "businessPhoneNo": "+2348134567890",
                    "email": "belly@fill.com",
                  },
                  "delivered": false,
                  "description": "",
                  "id": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
                  "imageURL": "http://www.gratednutmeg.com/wp-content/uploads/2015/03/DSC_07722.jpg",
                  "price": "2000.00",
                  "quantity": 2,
                  "title": "Adalu (Beans and Corn) and Local Fish ",
                  "vegetarian": false,
                },
              ]
            }
            type="user"
          />
          <div
            className="customer-details"
          >
            <p>
              Status:
              <span
                className="danger"
                style={
                  Object {
                    "textTransform": "capitalize",
                  }
                }
              >
                 
                canceled
              </span>
            </p>
            <p>
              Address Provided: 
              4, Church Street, Yaba
            </p>
            <p>
              Number Provided: 
              +2348134567890
            </p>
          </div>
        </React.Fragment>
      </div>
    </React.Fragment>
  </React.Fragment>
</View>
`;

exports[`OrderDetails renders connected component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CustomerOrderDetails)
    isFetching={false}
    match={
      Object {
        "params": Object {
          "id": "fb097bde-5959-45ff-8e21-51184fa60c25",
        },
      }
    }
    user={
      Object {
        "email": "emiola@olasanmi.com",
        "firstname": "Emiola",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "lastname": "Olasanmi",
        "role": "customer",
      }
    }
  >
    <CustomerOrderDetails
      fetchOrders={[Function]}
      isFetching={false}
      logout={[Function]}
      match={
        Object {
          "params": Object {
            "id": "fb097bde-5959-45ff-8e21-51184fa60c25",
          },
        }
      }
      order={
        Object {
          "createdAt": "2018-04-06T00:47:03.687Z",
          "deliveryAddress": "4, Church Street, Yaba",
          "deliveryPhoneNo": "+2348134567890",
          "id": "fb097bde-5959-45ff-8e21-51184fa60c25",
          "meals": Array [
            Object {
              "caterer": Object {
                "businessAddress": "4, Church Street, Yaba",
                "businessName": "FoodCircle",
                "businessPhoneNo": "+2348134567890",
                "email": "food@circle.com",
              },
              "delivered": true,
              "description": "Sharwama contains no animal products, perfect for dieters",
              "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
              "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
              "price": "1200.00",
              "quantity": 1,
              "title": "Vegetable Sharwama and Guava Smoothie",
              "vegetarian": true,
            },
          ],
          "status": "delivered",
          "updatedAt": "2018-04-06T00:47:03.687Z",
        }
      }
      user={
        Object {
          "email": "emiola@olasanmi.com",
          "firstname": "Emiola",
          "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
          "lastname": "Olasanmi",
          "role": "customer",
        }
      }
    >
      <View
        isFetching={false}
        logout={[Function]}
        showTime={true}
        type="customerOrderDetails"
        user={
          Object {
            "email": "emiola@olasanmi.com",
            "firstname": "Emiola",
            "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
            "lastname": "Olasanmi",
            "role": "customer",
          }
        }
      >
        <div
          className="user"
        >
          <Header
            active="customerOrderDetails"
            dateType="customerOrderDetails"
            logout={[Function]}
            showTime={true}
            type="customer"
            user={
              Object {
                "email": "emiola@olasanmi.com",
                "firstname": "Emiola",
                "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                "lastname": "Olasanmi",
                "role": "customer",
              }
            }
          >
            <div
              className="header"
            >
              <div
                className="navbar white-navbar"
              >
                <CustomerHeader
                  active="customerOrderDetails"
                  dateType="customerOrderDetails"
                  logout={[Function]}
                  showTime={true}
                  type="customer"
                  user={
                    Object {
                      "email": "emiola@olasanmi.com",
                      "firstname": "Emiola",
                      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                      "lastname": "Olasanmi",
                      "role": "customer",
                    }
                  }
                >
                  <div
                    className="page-title"
                  >
                    <h3>
                      <Link
                        href="/"
                        replace={false}
                        to="/"
                      >
                        <a
                          href="/"
                          onClick={[Function]}
                        >
                          BOOK-A-MEAL
                        </a>
                      </Link>
                    </h3>
                  </div>
                  <div
                    className="navlinks"
                  >
                    <div
                      className="user-info"
                    >
                      <Dropdown
                        content={
                          <UNDEFINED>
                            <Link
                              className="false"
                              href="/"
                              replace={false}
                              to="/"
                            >
                              Menu
                            </Link>
                            <Link
                              href="/"
                              replace={false}
                              to="/"
                            >
                              Order History
                            </Link>
                            <Link
                              href="/"
                              onClick={[Function]}
                              replace={false}
                              to="/"
                            >
                              Logout
                            </Link>
                          </UNDEFINED>
                        }
                        toggler={
                          <div
                            className="username-circle circle-click"
                          >
                            <p
                              className="circle-click"
                            >
                              E
                              â–¿
                            </p>
                          </div>
                        }
                        type="customer"
                      >
                        <div
                          className="dropdown"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        >
                          <Link
                            className="link"
                            clickHandler={[Function]}
                            id="dropdown-toggler"
                          >
                            <button
                              className="link-btn link"
                              id="dropdown-toggler"
                              onClick={[Function]}
                            >
                              <div
                                className="username-circle circle-click"
                              >
                                <p
                                  className="circle-click"
                                >
                                  E
                                  â–¿
                                </p>
                              </div>
                            </button>
                          </Link>
                          <div
                            className="dropdown-content"
                            id="dropdown-content"
                          >
                            <Link
                              className="false"
                              href="/"
                              replace={false}
                              to="/"
                            >
                              <a
                                className="false"
                                href="/"
                                onClick={[Function]}
                              >
                                Menu
                              </a>
                            </Link>
                            <Link
                              href="/"
                              replace={false}
                              to="/"
                            >
                              <a
                                href="/"
                                onClick={[Function]}
                              >
                                Order History
                              </a>
                            </Link>
                            <Link
                              href="/"
                              onClick={[Function]}
                              replace={false}
                              to="/"
                            >
                              <a
                                href="/"
                                onClick={[Function]}
                              >
                                Logout
                              </a>
                            </Link>
                          </div>
                        </div>
                      </Dropdown>
                      <p>
                        Â Â 
                        Emiola
                      </p>
                    </div>
                  </div>
                </CustomerHeader>
              </div>
            </div>
          </Header>
          <div
            className="main-wrapper"
            id="has-modal"
          >
            <div
              className="order-confirmation order-details"
            >
              <h2
                className="text-center"
                style={
                  Object {
                    "paddingTop": "1em",
                  }
                }
              >
                Order #
                fb097bde-5959-45ff-8e21-51184fa60c25
              </h2>
              <p
                className="order-date text-center"
              >
                70/01/01 12:00 am
              </p>
              <hr />
              <OrderSummary
                meals={
                  Array [
                    Object {
                      "caterer": Object {
                        "businessAddress": "4, Church Street, Yaba",
                        "businessName": "FoodCircle",
                        "businessPhoneNo": "+2348134567890",
                        "email": "food@circle.com",
                      },
                      "delivered": true,
                      "description": "Sharwama contains no animal products, perfect for dieters",
                      "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                      "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                      "price": "1200.00",
                      "quantity": 1,
                      "title": "Vegetable Sharwama and Guava Smoothie",
                      "vegetarian": true,
                    },
                  ]
                }
              >
                <div
                  className="order-summary"
                >
                  <div
                    key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
                  >
                    <p>
                      1
                      x
                    </p>
                    <p>
                      Vegetable Sharwama and Guava Smoothie
                    </p>
                    <p>
                      â‚¦
                      1200
                    </p>
                  </div>
                </div>
              </OrderSummary>
              <OrderAmount
                meals={
                  Array [
                    Object {
                      "caterer": Object {
                        "businessAddress": "4, Church Street, Yaba",
                        "businessName": "FoodCircle",
                        "businessPhoneNo": "+2348134567890",
                        "email": "food@circle.com",
                      },
                      "delivered": true,
                      "description": "Sharwama contains no animal products, perfect for dieters",
                      "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                      "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                      "price": "1200.00",
                      "quantity": 1,
                      "title": "Vegetable Sharwama and Guava Smoothie",
                      "vegetarian": true,
                    },
                  ]
                }
                type="user"
              >
                <div
                  className="order-amount false"
                >
                  <div>
                    <p>
                      Total
                    </p>
                    <h2>
                      â‚¦
                      1200
                    </h2>
                  </div>
                </div>
              </OrderAmount>
              <div
                className="customer-details"
              >
                <p>
                  Status:
                  <span
                    className="success"
                    style={
                      Object {
                        "textTransform": "capitalize",
                      }
                    }
                  >
                     
                    delivered
                  </span>
                </p>
                <p>
                  Address Provided: 
                  4, Church Street, Yaba
                </p>
                <p>
                  Number Provided: 
                  +2348134567890
                </p>
              </div>
            </div>
          </div>
          <Connect(Modal)>
            <Modal
              open={false}
              toggleModal={[Function]}
              type={null}
            />
          </Connect(Modal)>
          <Footer>
            <div
              className="footer"
            >
              <p>
                Â© 2018. Iveren Shaguy. All images were gotten from Google and are not mine.
              </p>
            </div>
          </Footer>
        </div>
      </View>
    </CustomerOrderDetails>
  </Connect(CustomerOrderDetails)>
</Provider>
`;

exports[`OrderDetails renders correctly when not fetching 1`] = `
<View
  isFetching={false}
  logout={[MockFunction]}
  showTime={true}
  type="customerOrderDetails"
  user={
    Object {
      "email": "emiola@olasanmi.com",
      "firstname": "Emiola",
      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
      "lastname": "Olasanmi",
      "role": "customer",
    }
  }
>
  <React.Fragment>
    <React.Fragment>
      <div
        className="order-confirmation order-details"
      >
        <h2
          className="text-center"
          style={
            Object {
              "paddingTop": "1em",
            }
          }
        >
          Order #
          fb097bde-5959-45ff-8e21-51184fa70c25
        </h2>
        <p
          className="order-date text-center"
        >
          70/01/01 12:00 am
        </p>
        <hr />
        <React.Fragment>
          <OrderSummary
            meals={
              Array [
                Object {
                  "caterer": Object {
                    "businessAddress": "4, Church Street, Yaba",
                    "businessName": "BellyFill",
                    "businessPhoneNo": "+2348134567890",
                    "email": "belly@fill.com",
                  },
                  "delivered": false,
                  "description": "",
                  "id": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
                  "imageURL": "http://www.gratednutmeg.com/wp-content/uploads/2015/03/DSC_07722.jpg",
                  "price": "2000.00",
                  "quantity": 2,
                  "title": "Adalu (Beans and Corn) and Local Fish ",
                  "vegetarian": false,
                },
              ]
            }
          />
          <OrderAmount
            meals={
              Array [
                Object {
                  "caterer": Object {
                    "businessAddress": "4, Church Street, Yaba",
                    "businessName": "BellyFill",
                    "businessPhoneNo": "+2348134567890",
                    "email": "belly@fill.com",
                  },
                  "delivered": false,
                  "description": "",
                  "id": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
                  "imageURL": "http://www.gratednutmeg.com/wp-content/uploads/2015/03/DSC_07722.jpg",
                  "price": "2000.00",
                  "quantity": 2,
                  "title": "Adalu (Beans and Corn) and Local Fish ",
                  "vegetarian": false,
                },
              ]
            }
            type="user"
          />
          <div
            className="customer-details"
          >
            <p>
              Status:
              <span
                className="danger"
                style={
                  Object {
                    "textTransform": "capitalize",
                  }
                }
              >
                 
                canceled
              </span>
            </p>
            <p>
              Address Provided: 
              4, Church Street, Yaba
            </p>
            <p>
              Number Provided: 
              +2348134567890
            </p>
          </div>
        </React.Fragment>
      </div>
    </React.Fragment>
  </React.Fragment>
</View>
`;

exports[`OrderDetails renders message when not fetching and that order doesn't exist 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CustomerOrderDetails)
    isFetching={false}
    match={
      Object {
        "params": Object {
          "id": "fb097bde-5959-45ff-8e21-51184fa61c25",
        },
      }
    }
    user={
      Object {
        "email": "emiola@olasanmi.com",
        "firstname": "Emiola",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "lastname": "Olasanmi",
        "role": "customer",
      }
    }
  >
    <CustomerOrderDetails
      fetchOrders={[Function]}
      isFetching={false}
      logout={[Function]}
      match={
        Object {
          "params": Object {
            "id": "fb097bde-5959-45ff-8e21-51184fa61c25",
          },
        }
      }
      order={null}
      user={
        Object {
          "email": "emiola@olasanmi.com",
          "firstname": "Emiola",
          "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
          "lastname": "Olasanmi",
          "role": "customer",
        }
      }
    >
      <View
        isFetching={false}
        logout={[Function]}
        showTime={true}
        type="customerOrderDetails"
        user={
          Object {
            "email": "emiola@olasanmi.com",
            "firstname": "Emiola",
            "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
            "lastname": "Olasanmi",
            "role": "customer",
          }
        }
      >
        <div
          className="user"
        >
          <Header
            active="customerOrderDetails"
            dateType="customerOrderDetails"
            logout={[Function]}
            showTime={true}
            type="customer"
            user={
              Object {
                "email": "emiola@olasanmi.com",
                "firstname": "Emiola",
                "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                "lastname": "Olasanmi",
                "role": "customer",
              }
            }
          >
            <div
              className="header"
            >
              <div
                className="navbar white-navbar"
              >
                <CustomerHeader
                  active="customerOrderDetails"
                  dateType="customerOrderDetails"
                  logout={[Function]}
                  showTime={true}
                  type="customer"
                  user={
                    Object {
                      "email": "emiola@olasanmi.com",
                      "firstname": "Emiola",
                      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                      "lastname": "Olasanmi",
                      "role": "customer",
                    }
                  }
                >
                  <div
                    className="page-title"
                  >
                    <h3>
                      <Link
                        href="/"
                        replace={false}
                        to="/"
                      >
                        <a
                          href="/"
                          onClick={[Function]}
                        >
                          BOOK-A-MEAL
                        </a>
                      </Link>
                    </h3>
                  </div>
                  <div
                    className="navlinks"
                  >
                    <div
                      className="user-info"
                    >
                      <Dropdown
                        content={
                          <UNDEFINED>
                            <Link
                              className="false"
                              href="/"
                              replace={false}
                              to="/"
                            >
                              Menu
                            </Link>
                            <Link
                              href="/"
                              replace={false}
                              to="/"
                            >
                              Order History
                            </Link>
                            <Link
                              href="/"
                              onClick={[Function]}
                              replace={false}
                              to="/"
                            >
                              Logout
                            </Link>
                          </UNDEFINED>
                        }
                        toggler={
                          <div
                            className="username-circle circle-click"
                          >
                            <p
                              className="circle-click"
                            >
                              E
                              â–¿
                            </p>
                          </div>
                        }
                        type="customer"
                      >
                        <div
                          className="dropdown"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        >
                          <Link
                            className="link"
                            clickHandler={[Function]}
                            id="dropdown-toggler"
                          >
                            <button
                              className="link-btn link"
                              id="dropdown-toggler"
                              onClick={[Function]}
                            >
                              <div
                                className="username-circle circle-click"
                              >
                                <p
                                  className="circle-click"
                                >
                                  E
                                  â–¿
                                </p>
                              </div>
                            </button>
                          </Link>
                          <div
                            className="dropdown-content"
                            id="dropdown-content"
                          >
                            <Link
                              className="false"
                              href="/"
                              replace={false}
                              to="/"
                            >
                              <a
                                className="false"
                                href="/"
                                onClick={[Function]}
                              >
                                Menu
                              </a>
                            </Link>
                            <Link
                              href="/"
                              replace={false}
                              to="/"
                            >
                              <a
                                href="/"
                                onClick={[Function]}
                              >
                                Order History
                              </a>
                            </Link>
                            <Link
                              href="/"
                              onClick={[Function]}
                              replace={false}
                              to="/"
                            >
                              <a
                                href="/"
                                onClick={[Function]}
                              >
                                Logout
                              </a>
                            </Link>
                          </div>
                        </div>
                      </Dropdown>
                      <p>
                        Â Â 
                        Emiola
                      </p>
                    </div>
                  </div>
                </CustomerHeader>
              </div>
            </div>
          </Header>
          <div
            className="main-wrapper"
            id="has-modal"
          >
            <p
              className="text-center"
            >
              This Order Does Not Exist
            </p>
          </div>
          <Connect(Modal)>
            <Modal
              open={false}
              toggleModal={[Function]}
              type={null}
            />
          </Connect(Modal)>
          <Footer>
            <div
              className="footer"
            >
              <p>
                Â© 2018. Iveren Shaguy. All images were gotten from Google and are not mine.
              </p>
            </div>
          </Footer>
        </div>
      </View>
    </CustomerOrderDetails>
  </Connect(CustomerOrderDetails)>
</Provider>
`;
